### PR TITLE
V3.4.0 to dev

### PR DESCRIPTION
**Pull request by simon-wang-gc**

### CODE REVIEW

**Comment by KingBain** [comment]
these commits are so large that there is no way to possible review and verify the code. What stands out to me is that we need to enable: 

(Easiest to hardwaresy levels of effort)
- Linting to implement coding standards. 
- Smaller releases to make manual code review possible
  - Switch to unit tests
- Terraform environment reconciliation. Backporting code changes will kill us

**Comment by simon-wang-gc** [comment]
Starting Terraform 1.7, unit tests are becoming available. So we can look into that for sure. https://developer.hashicorp.com/terraform/language/tests?product_intent=terraform

Upgrading projects is a prioritization question. Since our team has more resources, we can start doing regular upgrades for 100+ workspace we have in POC.

File: modules/v3.3.0/azure-storage-blob/alert.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # resource "azurerm_monitor_metric_alert" "datahub_proj_storage_alert" {
 #   count = var.storage_size_limit_tb > 0 ? 1 : 0
 
-#   name                = "${var.resource_prefix}-proj-storage-alert-${var.project_cd}-${var.environment_name}"
+#   name                = "${var.resource_prefix_alphanumeric}-proj-storage-alert-${var.project_cd}-${var.environment_name}"
```
Comment (yjmrobert) [inline modules/v3.3.0/azure-storage-blob/alert.tf:6]: Nice catch!!

**Review by yjmrobert — APPROVED** [review]

**Review by KingBain — APPROVED** [review]
LGTM, I have some things I can add here, but they're deeper downstream changes

**Pull request merged**